File: ios/LoverApp/Models/Person.swift
import Foundation

// Lightweight identity card for either user. Mirrors D.ME / D.PARTNER in data.js.
// Real production data is decrypted from `users` rows; this struct is the in-memory shape.

struct Person: Identifiable, Hashable {
    enum Tint: String, Codable { case rose, sage, amber }

    let id: String
    let name: String        // display name (will be decrypted from ciphertext in production)
    let initial: String     // single-letter or single-CJK avatar character
    let tint: Tint
}

extension Person {
    static let mockMe = Person(id: "kit", name: "Kit", initial: "K", tint: .rose)
    static let mockPartner = Person(id: "michel", name: "Michel", initial: "M", tint: .sage)
}

File: android/app/src/main/kotlin/michel/kit/us/domain/Person.kt
package michel.kit.us.domain

import androidx.compose.runtime.Immutable
import java.time.Instant

/**
 * Lightweight identity card for either user. Mirror of
 * ios/LoverApp/Models/Person.swift.
 *
 * Phase C — adds [lastSeenAt] sourced from `users.last_seen_at` on the
 * partner profile. Used by the chat header to render
 * "上次在線 X 分鐘前" when the realtime presence channel says the partner
 * is offline.
 */
@Immutable
data class Person(
    val id: String,
    val name: String,
    val initial: String,
    val tint: Tint,
    val lastSeenAt: Instant? = null
) {
    enum class Tint { rose, sage, amber }
}
